packages/augur-core/source/contracts/trading/FillOrder.sol

```diff
@@ -491,20 +493,22 @@ contract FillOrder is Initializable, ReentrancyGuard, IFillOrder {
 
     function fillOrder(address _filler, bytes32 _orderId, uint256 _amountFillerWants, bytes32 _tradeGroupId, bytes32 _fingerprint) external returns (uint256) {
         require(msg.sender == trade);
+        IERC20 _kycToken = storedContracts.orders.getKYCToken(_orderId);
+        require(_kycToken == IERC20(0) || _kycToken.balanceOf(_filler) > 0, "FillOrder.fillOrder: KYC token failure");
         Trade.Data memory _tradeData = Trade.create(storedContracts, _orderId, _filler, _amountFillerWants, _fingerprint);
         return fillOrderInternal(_filler, _tradeData, _amountFillerWants, _tradeGroupId);
     }
 
     function fillZeroXOrder(IMarket _market, uint256 _outcome, IERC20 _kycToken, uint256 _price, Order.Types _orderType, uint256 _amount, address _creator, bytes32 _tradeGroupId, bytes32 _fingerprint, address _filler) external returns (uint256) {
         require(msg.sender == zeroXTrade);
         require(augur.isKnownMarket(_market));
+        require(_kycToken == IERC20(0) || _kycToken.balanceOf(_filler) > 0, "FillOrder.fillOrder: KYC token failure");
```

Comment:
Re-entering is technically still possible here, but looks to be without effect, since this call happens after the ZeroXTrade code, after the actual 0x fill (for maker amount remaining) and before a few checks. The code being skipped is here:
```
        if (!creatorHasFundsForTrade(_order, _amount)) {
            return 0;
        }
        // If the maker is also the taker we also just skip the trade
        if (_order.makerAddress == _taker) {
            return 0;
        }
```

Which are just optimizations, so this looks good to me.